code/game/objects/effects/decals/Cleanable/misc.dm

```diff
@@ -38,11 +38,11 @@
 	icon_state = "dirt"
 	mouse_opacity = 0
 
-/obj/effect/decal/cleanable/Initialize(var/mapload, var/_age)
+/obj/effect/decal/cleanable/Initialize(var/mapload, var/_age, var/dirt)
 	.=..()
 	var/turf/simulated/our_turf = src.loc
-	if(our_turf && istype(our_turf) && our_turf.can_dirty && _age)
-		our_turf.dirt = 101
+	if(dirt && our_turf && istype(our_turf) && our_turf.can_dirty)
+		our_turf.dirt = dirt

```

Comment:
There is a small problem with this, though its kinda pre_existing. This means pre-mapped dirt will not properly dirty-up the tirf and will be stacking. I think potential solution would be to self-destruct if there is dirt on tile already; and also make sure that if we are NOT persistant dirt, we just apply maximum dirtiness to turf anyway.

I'd also  suggest making it clamp between dirt level at which it appears on tile and 101 , just in case

Finally, there is also fact that this will make icon updating for the dirt very awkward - it will start at maximum alpha like it already has anyway and then when someone walks over it will be updated for first time. Forcing update of alpha here is important.

Reply:
all suggestions applied - even better dirt code created

except that you can't just apply max dirt if not persistent, otherwise when the turf creates dirt for the very first time (when turf.dirt goes over 50), it'd just jump instantly to the max

Reply:
mapload might be a good way to handle that? Im not sure if its relevant for init or not. Or maybe another var that is set to false when creating it 'naturally' rather than pre-mapped? Something along those lines can work.